sktime/forecasting/naive.py

```diff
@@ -70,42 +79,40 @@ def fit(self, y_train, fh=None, X_train=None):
         self._set_oh(y_train)
         self._set_fh(fh)
 
-        if self.strategy in ("last", "seasonal_last"):
+        if self.strategy == "last":
             if self.window_length is not None:
-                warn("For the `last` and `seasonal_last` strategy, "
+                warn("For the `last` strategy, "
                      "the `window_length_` value will be ignored.")
 
-        if self.strategy in ("last", "mean"):
-            if self.sp is not None:
-                warn("For the `last` and `mean` strategy, "
-                     "the `sp` value will be ignored.")
-
-        if self.strategy == "last":
+        if self.strategy == "last" and self.sp is None:
             self.window_length_ = 1
 
-        if self.strategy == "seasonal_last":
-            if self.sp is None:
-                raise NotImplementedError(
-                    "Automatic estimation of the seasonal periodicity `sp` "
-                    "from the data is not implemented yet; "
-                    "please specify the `sp` value.")
+        if self.strategy == "last" and self.sp is not None:
             self.sp_ = check_sp(self.sp)
 
             # window length we need for forecasts is just the
             # length of seasonal periodicity
             self.window_length_ = self.sp_
 
         if self.strategy == "mean":
+            # check window length is greater than sp for seasonal mean
+            if self.window_length is not None and self.sp is not None:
+                if self.window_length < self.sp:
+                    param1, param2 = "window_length", "sp"
```

Comment:
why do you define param1, param2 here? Can we not just use the strings directly in the ValueError()?